### ISSUE TITLE
outline getting removed on resize

### ISSUE THREAD

**Issue by markusmoormann** [lovell/sharp#2408]
we have an issue with black outlines of images getting removed in the current version (0.26.1). In 0.24.1 we still had the outline, though it was more grey than black. 
What are the steps to reproduce?
```
sharp("test.jpg").resize({width: 1400}).toFile("test2.jpg")
```

What is the expected behaviour?
the outline is still visible after resizing the image
test image:
[
![test](https://user-images.githubusercontent.com/2950375/95856046-b4b5dd00-0d59-11eb-9cc8-0de0ac717410.jpg)
](url)

test result in 0.26.1
![test0 26](https://user-images.githubusercontent.com/2950375/95856261-09f1ee80-0d5a-11eb-878a-fa31e6dc80c7.jpg)

test result in 0.24.1:
![test2](https://user-images.githubusercontent.com/2950375/95856167-e2028b00-0d59-11eb-94e7-484786e79525.jpg)

any ideas?

**Comment by lovell** [comment]
Hi, did you see the `fastShrinkOnLoad` option?

https://sharp.pixelplumbing.com/api-resize#parameters

**Comment by markusmoormann** [comment]
unfortunately this doesn't work 
```sharp("test.jpg").resize({width: 1400,fastShrinkOnLoad:true}).toFile("test0.26_fast.jpg")```
![test0 26_fast](https://user-images.githubusercontent.com/2950375/95863725-d23c7400-0d64-11eb-8154-08080ecf3886.jpg)

**Comment by kleisauke** [comment]
I could reproduce this on libvips master with:
```bash
vips resize x.jpg tn_x.jpg 0.929615
```

This small `+ 0.5`-patch in the libvips mask calculations seems to resolve this.

<details>
 <summary>Patch</summary>

```diff
diff --git a/libvips/resample/templates.h b/libvips/resample/templates.h
index 1111111..2222222 100644
--- a/libvips/resample/templates.h
+++ b/libvips/resample/templates.h
@@ -329,7 +329,7 @@ calculate_coefficients_triangle( double *c,
 
 	sum = 0;
 	for( i = 0; i < n_points; i++ ) {
-		const double xp = (i - half) / shrink;
+		const double xp = (i - half + 0.5) / shrink;
 
 		double l;
 
@@ -367,7 +367,7 @@ calculate_coefficients_cubic( double *c,
 
 	sum = 0;
 	for( i = 0; i < n_points; i++ ) {
-		const double xp = (i - half) / shrink;
+		const double xp = (i - half + 0.5) / shrink;
 		const double axp = VIPS_FABS( xp ); 
 		const double axp2 = axp * axp;
 		const double axp3 = axp2 * axp;
@@ -416,7 +416,7 @@ calculate_coefficients_lanczos( double *c,
 
 	sum = 0;
 	for( i = 0; i < n_points; i++ ) {
-		const double xp = (i - half) / shrink;
+		const double xp = (i - half + 0.5) / shrink;
 
 		double l;
 
```
</details>

(I'm not sure if this patch could cause regressions, I'll have to check this against [this](https://github.com/kleisauke/vips-issue-703) and [this](https://github.com/kleisauke/pyvips-issue-148) repository)

I now see:
![tn_x](https://user-images.githubusercontent.com/12746591/95872806-83e0a280-0d6f-11eb-83ac-6a6d164545e5.jpg)

/cc @jcupitt as he recently did https://github.com/libvips/libvips/pull/1647.

**Comment by kleisauke** [comment]
With commit https://github.com/kleisauke/libvips/commit/18eebf7a9491163d72e07f9c674a9af1c9517a90 (which is included in PR https://github.com/libvips/libvips/pull/1769), I now see:
<details>
 <summary>Output image</summary>

![tn_x](https://user-images.githubusercontent.com/12746591/95912122-ac35c480-0da2-11eb-87ba-ff0e30409840.jpg)

</details>

Which looks to be the correct output for this image. This change causes no regressions on the above linked repositories. It also seems to improve the 9.4x resize on the [pyvips-issue-148](https://github.com/kleisauke/pyvips-issue-148) repository, see:
| VIPS before | VIPS after |
|  :---: |  :---: |
| ![8x_vips_resize.png](https://raw.githubusercontent.com/kleisauke/pyvips-issue-148/f4b7c05187ec20ff73787ef1aa6cfe4b1776e5b3/output-patch/9.4x_vips_resize.png) | ![8x_vips_resize.png](https://raw.githubusercontent.com/kleisauke/pyvips-issue-148/f6841e1a538d9c8a661d968ed33f59874ca2efbb/output-patch/9.4x_vips_resize.png) |
| Above border is missing | No missing borders |

**Comment by markusmoormann** [comment]
This sounds great! How ist your process now? will there be a bugfix version?

**Comment by lovell** [comment]
@markusmoormann Please subscribe to https://github.com/libvips/libvips/pull/1769 for updates.

**Comment by lovell** [comment]
This is fixed in libvips 8.10.4, which will be included in the prebuilt binaries provided by sharp v0.27.0.

**Comment by gaoshun** [comment]
Great! @lovell when will v0.27.0 be released?

**Comment by lovell** [comment]
sharp v0.27.0 now available with a prebuilt libvips v8.10.5 - thanks for reporting this.

**Comment by markusmoormann** [comment]
@lovell works flawlessly. Thanks
